### ISSUE TITLE
helm-wrapper randomly failing to decrypt successfully

### ISSUE THREAD

**Issue by djablonski** [futuresimple/helm-secrets#55]
When usinf `helm-wrapper` to decrypt files, we randomly get errors about the decrypted file not being found:

```
>>>>>> Decrypt
Decrypting /tmp/tmp.BdvUTnv8Ji/config-server/secrets.yaml

Release "config-server" does not exist. Installing it now.
Error: open /tmp/tmp.BdvUTnv8Ji/config-server/secrets.yaml.dec: no such file or directory
```

The general setup works fine, since there are other charts being deployed the exact same way in parallel where it works fine, and there is also no pattern in where its failing, as far as I can see. Sometimes its this chart, sometimes another and other times none at all... :(

Helm version: v2.9.1
Helm secrets version: 1.3.0
Sops version: 3.0.3

Any proposals appreciated!

**Comment by szibis** [comment]
Maybe parallelism is the key to find the answer. Maybe one chart removes temp secrets file for another and we need to scope temp file to a specific chart.

**Comment by djablonski** [comment]
Indeed, I have now narrowed this down to being a concurrency issue. But still not yet investigated deeply where the root cause lies.

**Comment by djablonski** [comment]
UPDATE: The issue is caused by `clean` deleting all `.ced` files in the tree, no matter if that tree is a chart or not.
In out use case, we have a tree of _override values_, consisting of paths like `.../<environment>/<chart>/<overfide-values-file` where those `<override-values-file`s might be secrets also. We're creating a commandline for `helm-wrapper` which looks like this:

```
helm-wrapper install <chart-name> --values .../<environment>/<chart>/secrets.yaml
```

Here the `cleanup task is going wild, since it considers `.../environment/` as the base dir, and deletes every `.dec` file in all the subdirs. Since we have multiple commands running in parallel for different charts, whose files are erroneously get deleted, then just fail.

Looking into a possible solution tomorrow, and maybe creating a PR to solve this...
